### DISCUSSION TITLE
Hosting a Wasm app in a native shell

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#10668] [Q&A] [answered]
[_From Discord_:](https://discord.com/channels/372137812037730304/550416151172087808/1049452218954293279)

Hey folks, silly question. Let's say I'm doing a Uno Wasm app. Just for conveniency of being able to install the app without having to pass through PWA, would it make sense to make blank app (WASDK on Windows, Uno Mac Catalyst or even Swift on MacOS, or Electron...) that just has a WebView that shows the Uno Wasm app, or is it overkill / useless / terrible idea?

**Comment by jeromelaban** [comment, marked as answer]
It's an interesting question. Technically, it's possible to implement such a scenario. I don't know if recent updates of native Web Views have less restrictions with performance. For a while, Javascript jitting was disabled in apps and WebAssembly may as well be disabled.

That being said, most stores don't want "web wrapper"-only apps, and you may not be able to deploy those on stores, unless you're using MDM scenarios.

You may also have to implement interop scenarios with platform features, which are not necessarily easy to build

**Reply by veler** [reply]
That makes sense. Sounds like it's fairly easy to publish a PWA to the App Store though. :) https://blog.pwabuilder.com/posts/publish-your-pwa-to-the-ios-app-store/

**Comment by DierkDroth** [comment]
@veler I'm curious: what business value would see in packaging a web app for deployment in a store?

I rather would be trying the opposite and do 'web deployment' rather than store deployment to avoid the restrictions coming with the store deployment as well as avoiding revenue sharing with the store provider. The only benefit of store deployment which I could see is some marketing benefit, which needed to be weighed against the downsides...

What am I missing?
